MapboxNavigation/NavigationViewController.swift

```diff
@@ -259,7 +273,7 @@ public class NavigationViewController: NavigationPulleyViewController, RouteMapV
         
         super.init(contentViewController: mapViewController, drawerViewController: tableViewController)
         
-        self.styles = styles ?? [DefaultStyle()]
+        self.styles = styles ?? [styleForSunPosition]
```

Comment:
I'd suggest we pass both DayStyle and NightStyle (so we have a ref if the sun sets while in navigation) but later when the style is applied, we'd filter out non-appropriate style. Adding a `dayTime` enum property (day/night) on `Style` could help.